File: 20. Valid Parentheses/20. Valid Parentheses.ts
const isValid = (s: string): boolean => {
  const stack = [];
  const result = [...s].every((element) => {
    let pop = true;
    switch (element) {
      case "(":
      case "[":
      case "{":
        stack.push(element);
        break;
      case ")":
        pop = stack.pop() === "(" ? true : false;
        break;
      case "]":
        pop = stack.pop() === "[" ? true : false;
        break;
      case "}":
        pop = stack.pop() === "{" ? true : false;
        break;
    }

    return pop;
  });

  if (stack.length) return false;
  return result;
};

File: 20. Valid Parentheses/20. Valid Parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>();
        char[] ch = s.toCharArray();
        for(char chr: ch){
            switch(chr){
                case '[':
                case '(':
                case '{':
                    stack.push(chr);
                    break;
                case ']':
                    if(stack.isEmpty() || stack.pop() != '[') return false;
                    break;
                case ')':
                    if(stack.isEmpty() || stack.pop() != '(') return false;
                    break;
                case '}':
                    if(stack.isEmpty() || stack.pop() != '{') return false;
                    break;
            }
        }
        
        if(!stack.isEmpty()) return false;
        return true;            
    }
}